apps/backoffice-v2/src/pages/Statistics/components/PortfolioRiskStatistics/hooks/usePortfolioRiskStatisticsLogic/usePortfolioRiskStatisticsLogic.tsx

```diff
@@ -5,82 +5,74 @@ import {
   riskLevelToBackgroundColor,
   riskLevelToFillColor,
 } from '@/pages/Statistics/components/PortfolioRiskStatistics/constants';
+import { StatisticsOutputSchema } from '@/pages/Statistics/statistics.query';
+import { z } from 'zod';
 
-export const usePortfolioRiskStatisticsLogic = () => {
+export const usePortfolioRiskStatisticsLogic = ({
+  data,
+}: {
+  data: z.infer<typeof StatisticsOutputSchema>;
+}) => {
   const [parent] = useAutoAnimate<HTMLTableSectionElement>();
-  const [sorting, setSorting] = useState<SortDirection>('desc');
-  const onSort = useCallback(
+  const [riskIndicatorsSorting, setRiskIndicatorsSorting] = useState<SortDirection>('desc');
+  const onSortRiskIndicators = useCallback(
     (sort: SortDirection) => () => {
-      setSorting(sort);
+      setRiskIndicatorsSorting(sort);
     },
     [],
   );
-  const sortedData = useMemo(
-    () =>
-      [
-        { riskType: 'Scam and fraud', amount: 15 },
-        { riskType: 'IP Rights Infringement', amount: 12 },
-        { riskType: 'Missing Terms and Conditions', amount: 10 },
-        { riskType: 'Counterfeit Goods', amount: 8 },
-        { riskType: 'Sanctions', amount: 4 },
-      ]
-        ?.slice()
-        .sort((a, b) => {
-          if (sorting === 'asc') {
-            return a.amount - b.amount;
-          }
-
-          return b.amount - a.amount;
-        }),
-    [sorting],
-  );
+  const totalRiskIndicators = data.riskIndicators.reduce((acc, curr) => acc + curr.count, 0);
+  const riskIndicators = useMemo(() => {
+    return data.riskIndicators
+      .sort((a, b) => {
+        if (riskIndicatorsSorting === 'asc') {
+          return a.count - b.count;
+        }
+        return b.count - a.count;
+      })
+      .slice(0, 5);
+  }, [data, riskIndicatorsSorting]);
   const widths = useMemo(() => {
-    const maxValue = Math.max(...sortedData.map(item => item.amount), 0);
+    const maxValue = Math.max(...riskIndicators.map(item => item.count), 0);
 
-    return sortedData.map(item =>
-      item.amount === 0 ? 0 : Math.max((item.amount / maxValue) * 100, 2),
+    return riskIndicators.map(item =>
+      item.count === 0 ? 0 : Math.max((item.count / maxValue) * 100, 2),
     );
-  }, [sortedData]);
+  }, [riskIndicators]);
   const filters = [
     {
       name: 'Merchant Monitoring',
```

Comment:
The merchant monitoring navigation link is not from a filter by the way. Since `entity` right now represents `individuals` or `businesses` we might want to reconsider the `entityPlural` approach. Also, are you familiar with this package? https://www.npmjs.com/package/pluralize

Reply:
I am familiar with `pluralize`, but I don't really need it here. The term is always plural, so when we add it to the JSON, it should already be pluralized. Why add another package to the runtime when I can just type the right entity?

Reply:
Adding a new column to filters for a very specific use case not directly related to how filters are used is something I don't want to do if I don't have to. `entity` and `entityPlural` can be different words entirely for better and worse.

Reply:
Sorry, I have to move on. I'll merge the PR for now but we will revisit this topic later if you want.